x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -1303,6 +1303,7 @@ export class RulesClient {
             ruleId: id,
             ruleName: attributes.name,
             ruleType: this.ruleTypeRegistry.get(attributes.alertTypeId),
+            executionId: 'rules-client', // TODO: Is it possible to get executionId from relevant `TaskRunner` instance, or does it need to saved to `params` of AlertTaskInstance?
```

Comment:
@pmuellr @mikecote -- I was able to thread `executionId` most everywhere it needed to be by adding it as a private readonly property of [task_runner](https://github.com/elastic/kibana/pull/113058/files#diff-27ff694dc534e9e509d43b00c563eabada722db2371c9bcf016ff3c8b929c5fbR97), however I'm not sure of a way to fetch it from this context within `rules_client` (for disabled `recovered-instance`). Perhaps I'm missing something here, but seems like it might have to be stored on `params` of AlertTaskInstance? Thoughts?

Reply:
The code this is pointing to is in the `disableWithOCC()` function.  We started writing out "recovered" events for active alerts when a rule is disabled, otherwise some alerts can become permanently "sticky".

As such, there is no execution associated with these events.  I think it would be best for now, if we just don't provide a value for the field.  

I believe we're going to have to do something more, eventually, with these specific events - like adding a reason field we could populate with "recovered due to rule disabled" kind of thing.  But I think there were some other similar cases, so we're deferring providing more info till we need to and understand what kind of shape we need.

Reply:
Thanks @pmuellr! I'll keep `executionId` as optional in `CreateAlertEventLogRecordParams` then in support of writing events like this that aren't associated to a specific execution.